com.hazelcast.jet.sql.impl.connector.file.SqlAvroTest.when_conversionFails_then_queryFails

_4.5-maintenance_ (commit 32b32b4d36c4a0f8b9db81557e9a0a3eadd742fc)

Failed on Open JDK 8 (Zulu): http://jenkins.hazelcast.com/job/jet-oss-maintenance-zulu-jdk8/287/testReport/com.hazelcast.jet.sql.impl.connector.file/SqlAvroTest/when_conversionFails_then_queryFails/

Stacktrace:
```
java.lang.AssertionError: 

Expecting:
 <"The Jet SQL job failed: Exception in ProcessorTasklet{0631-d439-bfc8-0001/Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])#2}: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:379)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:347)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:55)
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.cannotParseError(AbstractStringConverter.java:177)
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.asInt(AbstractStringConverter.java:81)
	at com.hazelcast.sql.impl.type.converter.IntegerConverter.convertToSelf(IntegerConverter.java:97)
	at com.hazelcast.sql.impl.type.QueryDataType.convert(QueryDataType.java:150)
	at com.hazelcast.jet.sql.impl.extract.AvroQueryTarget.lambda$createFieldExtractor$1(AvroQueryTarget.java:49)
	at com.hazelcast.jet.sql.impl.connector.RowProjector.get(RowProjector.java:91)
	at com.hazelcast.sql.impl.expression.ColumnExpression.eval(ColumnExpression.java:68)
	at com.hazelcast.jet.sql.impl.ExpressionUtil.evaluate(ExpressionUtil.java:154)
	at com.hazelcast.jet.sql.impl.connector.RowProjector.project(RowProjector.java:83)
	at com.hazelcast.function.BiFunctionEx.apply(BiFunctionEx.java:48)
	at com.hazelcast.jet.core.processor.Processors.lambda$mapUsingServiceP$26d41dd1$1(Processors.java:873)
	at com.hazelcast.jet.function.TriFunction.apply(TriFunction.java:47)
	at com.hazelcast.jet.impl.processor.TransformUsingServiceP.tryProcess(TransformUsingServiceP.java:63)
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187)
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:591)
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:109)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:421)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:296)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:291)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:273)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:367)
	... 3 more
">
to end with:
 <"Cannot parse VARCHAR value to INTEGER">

	at com.hazelcast.jet.sql.impl.connector.file.SqlAvroTest.when_conversionFails_then_queryFails(SqlAvroTest.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
2021-05-16 17:59:45,325 [DEBUG] [when_conversionFails_then_queryFails] [c.h.j.s.i.JetSqlBackend]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Before logical opt:
LogicalProject(string=[$0])
  LogicalTableScan(table=[[hazelcast, public, o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2[projects=[0]]]])

2021-05-16 17:59:45,326 [DEBUG] [when_conversionFails_then_queryFails] [c.h.j.s.i.JetSqlBackend]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] After logical opt:
FullScanLogicalRel(table=[[hazelcast, public, o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2[projects=[0]]]])

2021-05-16 17:59:45,327 [DEBUG] [when_conversionFails_then_queryFails] [c.h.j.s.i.JetSqlBackend]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] After physical opt:
FullScanPhysicalRel(table=[[hazelcast, public, o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2[projects=[0]]]])

2021-05-16 17:59:45,330 [ INFO] [hz.youthful_robinson.cached.thread-18] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Starting job 0631-d439-bfc8-0001 based on submit request
2021-05-16 17:59:45,332 [ INFO] [hz.youthful_robinson.cached.thread-18] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Didn't find any snapshot to restore for job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001
2021-05-16 17:59:45,332 [ INFO] [hz.youthful_robinson.cached.thread-18] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Start executing job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001, execution graph in DOT format:
digraph DAG {
	"File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2]" [localParallelism=4];
	"Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])" [localParallelism=72];
	"ClientSink" [localParallelism=1];
	"File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2]" -> "Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])" [label="isolated", queueSize=1024];
	"Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])" -> "ClientSink" [label="distributed to [127.0.0.1]:5701-partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
2021-05-16 17:59:45,332 [DEBUG] [hz.youthful_robinson.cached.thread-18] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Building execution plan for job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001
2021-05-16 17:59:45,332 [DEBUG] [hz.youthful_robinson.cached.thread-18] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Built execution plans for job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001
2021-05-16 17:59:45,333 [DEBUG] [hz.youthful_robinson.cached.thread-18] [c.h.j.i.o.InitExecutionOperation]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Initializing execution plan for job 0631-d439-bfc8-0001, execution 0631-d439-bfc9-0001 from [127.0.0.1]:5701
2021-05-16 17:59:45,336 [ INFO] [hz.youthful_robinson.cached.thread-18] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Execution plan for jobId=0631-d439-bfc8-0001, jobName='0631-d439-bfc8-0001', executionId=0631-d439-bfc9-0001 initialized
2021-05-16 17:59:45,338 [DEBUG] [hz.youthful_robinson.cached.thread-63] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Init of job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001 was successful
2021-05-16 17:59:45,338 [DEBUG] [hz.youthful_robinson.cached.thread-63] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Executing job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001
2021-05-16 17:59:45,338 [ INFO] [hz.youthful_robinson.cached.thread-63] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Start execution of job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001 from coordinator [127.0.0.1]:5701
2021-05-16 17:59:45,342 [ WARN] [hz.youthful_robinson.jet.cooperative.thread-20] [c.h.j.i.e.TaskletExecutionService]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Exception in ProcessorTasklet{0631-d439-bfc8-0001/Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])#2}
com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:55) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.cannotParseError(AbstractStringConverter.java:177) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.asInt(AbstractStringConverter.java:81) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.IntegerConverter.convertToSelf(IntegerConverter.java:97) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.QueryDataType.convert(QueryDataType.java:150) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.sql.impl.extract.AvroQueryTarget.lambda$createFieldExtractor$1(AvroQueryTarget.java:49) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.RowProjector.get(RowProjector.java:91) ~[classes/:?]
	at com.hazelcast.sql.impl.expression.ColumnExpression.eval(ColumnExpression.java:68) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.sql.impl.ExpressionUtil.evaluate(ExpressionUtil.java:154) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.RowProjector.project(RowProjector.java:83) ~[classes/:?]
	at com.hazelcast.function.BiFunctionEx.apply(BiFunctionEx.java:48) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.core.processor.Processors.lambda$mapUsingServiceP$26d41dd1$1(Processors.java:873) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.function.TriFunction.apply(TriFunction.java:47) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.processor.TransformUsingServiceP.tryProcess(TransformUsingServiceP.java:63) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:591) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:109) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:421) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:296) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:291) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:273) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:367) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) [?:1.8.0_265]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:347) [hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
2021-05-16 17:59:45,344 [DEBUG] [hz.youthful_robinson.jet.cooperative.thread-28] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Execution of job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001 completed with failure
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{0631-d439-bfc8-0001/Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])#2}: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292) ~[?:1.8.0_265]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308) ~[?:1.8.0_265]
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:607) ~[?:1.8.0_265]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591) ~[?:1.8.0_265]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_265]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990) ~[?:1.8.0_265]
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:59) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$ExecutionTracker.taskletDone(TaskletExecutionService.java:468) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.dismissTasklet(TaskletExecutionService.java:399) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:385) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) [?:1.8.0_265]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:347) [hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
Caused by: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{0631-d439-bfc8-0001/Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])#2}: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:379) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	... 3 more
Caused by: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:55) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.cannotParseError(AbstractStringConverter.java:177) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.asInt(AbstractStringConverter.java:81) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.IntegerConverter.convertToSelf(IntegerConverter.java:97) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.QueryDataType.convert(QueryDataType.java:150) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.sql.impl.extract.AvroQueryTarget.lambda$createFieldExtractor$1(AvroQueryTarget.java:49) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.RowProjector.get(RowProjector.java:91) ~[classes/:?]
	at com.hazelcast.sql.impl.expression.ColumnExpression.eval(ColumnExpression.java:68) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.sql.impl.ExpressionUtil.evaluate(ExpressionUtil.java:154) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.RowProjector.project(RowProjector.java:83) ~[classes/:?]
	at com.hazelcast.function.BiFunctionEx.apply(BiFunctionEx.java:48) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.core.processor.Processors.lambda$mapUsingServiceP$26d41dd1$1(Processors.java:873) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.function.TriFunction.apply(TriFunction.java:47) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.processor.TransformUsingServiceP.tryProcess(TransformUsingServiceP.java:63) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:591) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:109) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:421) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:296) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:291) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:273) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:367) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	... 3 more
2021-05-16 17:59:45,367 [DEBUG] [ForkJoinPool.commonPool-worker-57] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Execution of job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001 has failures: [[127.0.0.1]:5701=com.hazelcast.jet.JetException: Exception in ProcessorTasklet{0631-d439-bfc8-0001/Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])#2}: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER]
2021-05-16 17:59:45,367 [DEBUG] [hz.youthful_robinson.cached.thread-45] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Sending CompleteExecutionOperation for job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001
2021-05-16 17:59:45,367 [DEBUG] [hz.youthful_robinson.cached.thread-45] [c.h.j.i.o.CompleteExecutionOperation]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Completing execution 0631-d439-bfc9-0001 from caller [127.0.0.1]:5701, error=com.hazelcast.jet.JetException: Exception in ProcessorTasklet{0631-d439-bfc8-0001/Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])#2}: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
2021-05-16 17:59:45,367 [DEBUG] [hz.youthful_robinson.cached.thread-45] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Completed execution of job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001
2021-05-16 17:59:45,370 [ERROR] [hz.youthful_robinson.cached.thread-42] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Execution of job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001 failed
	Start time: 2021-05-16T17:59:45.330
	Duration: 00:00:00.040
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
com.hazelcast.jet.JetException: Exception in ProcessorTasklet{0631-d439-bfc8-0001/Project(File[public.o_d10c8f7a_9486_43e9_8aac_1ae601aa07e2])#2}: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:379) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_265]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:347) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_265]
Caused by: com.hazelcast.sql.impl.QueryException: Cannot parse VARCHAR value to INTEGER
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:55) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.cannotParseError(AbstractStringConverter.java:177) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.AbstractStringConverter.asInt(AbstractStringConverter.java:81) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.converter.IntegerConverter.convertToSelf(IntegerConverter.java:97) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.sql.impl.type.QueryDataType.convert(QueryDataType.java:150) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.sql.impl.extract.AvroQueryTarget.lambda$createFieldExtractor$1(AvroQueryTarget.java:49) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.RowProjector.get(RowProjector.java:91) ~[classes/:?]
	at com.hazelcast.sql.impl.expression.ColumnExpression.eval(ColumnExpression.java:68) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.sql.impl.ExpressionUtil.evaluate(ExpressionUtil.java:154) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.connector.RowProjector.project(RowProjector.java:83) ~[classes/:?]
	at com.hazelcast.function.BiFunctionEx.apply(BiFunctionEx.java:48) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.core.processor.Processors.lambda$mapUsingServiceP$26d41dd1$1(Processors.java:873) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.function.TriFunction.apply(TriFunction.java:47) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.processor.TransformUsingServiceP.tryProcess(TransformUsingServiceP.java:63) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:591) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:109) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:421) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:296) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:291) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:273) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:367) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	... 3 more
2021-05-16 17:59:45,371 [DEBUG] [hz.youthful_robinson.cached.thread-42] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001 is completed
2021-05-16 17:59:45,380 [ERROR] [hz.youthful_robinson.async.thread-1] [c.h.j.i.o.TerminateExecutionOperation]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] job 0631-d439-bfc8-0001, execution 0631-d439-bfc9-0001 not found for coordinator [127.0.0.1]:5701 for 'TerminateExecutionOperation'
com.hazelcast.jet.core.TopologyChangedException: job 0631-d439-bfc8-0001, execution 0631-d439-bfc9-0001 not found for coordinator [127.0.0.1]:5701 for 'TerminateExecutionOperation'
	at com.hazelcast.jet.impl.JobExecutionService.assertExecutionContext(JobExecutionService.java:326) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.operation.TerminateExecutionOperation.run(TerminateExecutionOperation.java:59) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:272) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:600) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:579) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:240) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) ~[hazelcast-4.2.jar:4.2]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:264) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:247) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$cancelExecutionInvocations$16(MasterJobContext.java:601) ~[hazelcast-jet-core-4.5.1-SNAPSHOT.jar:4.5.1-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_265]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) [hazelcast-4.2.jar:4.2]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [hazelcast-4.2.jar:4.2]
2021-05-16 17:59:45,380 [ERROR] [ForkJoinPool.commonPool-worker-51] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] job '0631-d439-bfc8-0001', execution 0631-d439-bfc9-0001: some TerminateExecutionOperation invocations failed, execution might remain stuck: [MemberInfo{address=[127.0.0.1]:5701, uuid=d9818a01-8125-4c46-8551-248d2d49ecea, liteMember=false, memberListJoinVersion=1}=com.hazelcast.jet.core.TopologyChangedException: job 0631-d439-bfc8-0001, execution 0631-d439-bfc9-0001 not found for coordinator [127.0.0.1]:5701 for 'TerminateExecutionOperation']
2021-05-16 17:59:48,107 [DEBUG] [hz.youthful_robinson.cached.thread-74] [c.h.j.i.JobRepository]: [127.0.0.1]:5701 [jet] [4.5.1-SNAPSHOT] Job cleanup took 6ms
2021-05-16 17:59:50,080 [ WARN] [hz.youthful_robinson.cached.thread-74] [c.h.j.i.JobRepository]: Logging took 1973 ms.
2021-05-16 17:59:50,085 [ INFO] [Time-limited test] [c.h.j.SimpleTestInClusterSupport]: Ditching 4 jobs in SimpleTestInClusterSupport.@After: [0631-d439-bfc8-0001, 0631-d439-bfc2-0001, 0631-d439-bfc4-0001, 0631-d439-bfc6-0001]
2021-05-16 17:59:50,145 [ INFO] [Time-limited test] [c.h.j.SimpleTestInClusterSupport]: Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:replicatedMapService/__sql.catalog]
BuildInfo right after when_conversionFails_then_queryFails(com.hazelcast.jet.sql.impl.connector.file.SqlAvroTest): BuildInfo{version='4.2', build='20210324', buildNumber=20210324, revision=405cfd1, enterprise=false, serializationVersion=1, jet=JetBuildInfo{version='4.5.1-SNAPSHOT', build='20210516', revision='32b32b4'}}
Hiccups measured while running test 'when_conversionFails_then_queryFails(com.hazelcast.jet.sql.impl.connector.file.SqlAvroTest):'
17:59:45, accumulated pauses: 1066 ms, max pause: 268 ms, pauses over 1000 ms: 0
17:59:50, accumulated pauses: 13 ms, max pause: 4 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlAvroTest.java
+++ b/hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlAvroTest.java
@@ -240,7 +240,7 @@ public class SqlAvroTest extends SqlTestSupport {
         );
 
         assertThatThrownBy(() -> sqlService.execute("SELECT * FROM " + name).iterator().hasNext())
-                .hasMessageEndingWith("Cannot parse VARCHAR value to INTEGER");
+                .hasMessageContaining("Cannot parse VARCHAR value to INTEGER");
     }
 
     @Test

--- a/hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlCsvTest.java
+++ b/hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlCsvTest.java
@@ -235,7 +235,7 @@ public class SqlCsvTest extends SqlTestSupport {
         );
 
         assertThatThrownBy(() -> sqlService.execute("SELECT * FROM " + name).iterator().hasNext())
-                .hasMessageEndingWith("Cannot parse VARCHAR value to INTEGER");
+                .hasMessageContaining("Cannot parse VARCHAR value to INTEGER");
     }
 
     @Test

--- a/hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlJsonTest.java
+++ b/hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlJsonTest.java
@@ -240,7 +240,7 @@ public class SqlJsonTest extends SqlTestSupport {
         );
 
         assertThatThrownBy(() -> sqlService.execute("SELECT * FROM " + name).iterator().hasNext())
-                .hasMessageEndingWith("Cannot parse VARCHAR value to INTEGER");
+                .hasMessageContaining("Cannot parse VARCHAR value to INTEGER");
     }
 
     @Test

